src/librustc/ty/constness.rs

```diff
@@ -55,7 +55,7 @@ impl<'a, 'tcx> TyCtxt<'a, 'tcx, 'tcx> {
             }
             // in order for a libstd function to be considered min_const_fn
             // it needs to be stable and have no `rustc_const_unstable` attribute
-            match self.lookup_stability(def_id) {
+            self.is_const_fn_raw(def_id) && match self.lookup_stability(def_id) {
```

Comment:
That doesn't work, because the intrinsics do not have "const" in their signature. ... Which makes me wonder why not. I'll see if I can get intrinsics to be marked as const at their definition, so we can simplify all this.

Reply:
I'm gonna start a separate PR for that and rebase this PR over the new one.